boskos/README.md

```diff
@@ -0,0 +1,131 @@
+# WIP boskos
+
+
+## Background
+[βοσκός](https://en.wiktionary.org/wiki/%CE%B2%CE%BF%CF%83%CE%BA%CF%8C%CF%82) - shepherd in greek!
+
+boskos is a resource manager service, that handles and manages different kind of resources and transition between different states.
+
+## Introduction
+
+Boskos is inited with a config of resources, one JSON entry per line, from `-config` flag
+
+A resource object looks like
+```
+type Resource struct {
+	Type       string    `json:"type"`
+	Name       string    `json:"name"`
+	State      string    `json:"state"`
+	Owner      string    `json:"owner"`
+	LastUpdate time.Time `json:"lastupdate"`
+}
+```
+
+Type can be GCPProject, cluster, or even a dota2 server, anything that you want to be a group of resources.
+Name is a unique identifier of the resource.
+State is a string that tells the current status of the resource.
+
+## API
+
+1. 	URL: /start
+	Desc: Use /start when you want to get hold of some resource.
+	Method: POST
+	URL Params: 
+		Required: type=[string] : type of requested resource
+		Required: state=[string] : state of the requested resource
+		Required: owner=[string] : requester of the resource
+	Return: error code or 200 with a valid Resource JSON object.
+	Example: /start?type=project&state=free&owner=senlu
+
+2.	URL: /done
+	Desc: use /done when you finish use some resource.
+	Method: POST
+	URL Params:
+		Required: name=[string] : name of finished resource
+		Required: state=[string] : dest state
+	Return: status code
+	Example: /done?name=k8s-jkns-foo&state=dirty
+
+3.	URL: /update
```

Comment:
emm this one is more like "heartbeat my timestamp", extend would probably need a duration associate with it.